application_form/services/offer.py

```diff
@@ -1,28 +1,35 @@
 from django.contrib.auth import get_user_model
 from django.core.exceptions import ValidationError
 from django.db import transaction
+from django.db.models import Q
 from django.utils import timezone
 
+from apartment.elastic.queries import get_apartment, get_apartment_uuids
 from apartment_application_service.utils import update_obj
-from application_form.enums import ApartmentReservationState, OfferState
+from application_form.enums import (
+    ApartmentReservationCancellationReason,
+    ApartmentReservationState,
+    OfferState,
+)
 from application_form.models import ApartmentReservation, Offer
+from application_form.services.application import cancel_reservation
 
 User = get_user_model()
 
 
 @transaction.atomic()
 def create_offer(offer_data: dict, user: User = None) -> Offer:
-    if hasattr(offer_data["apartment_reservation"], "offer"):
+    apartment_reservation = offer_data["apartment_reservation"]
+    if hasattr(apartment_reservation, "offer"):
         raise ValidationError(
             f"Cannot create an offer to reservation "
             f"{offer_data['apartment_reservation'].id} "
             f"because it already has an offer."
         )
 
     offer = Offer.objects.create(**offer_data)
-    offer_data["apartment_reservation"].set_state(
-        ApartmentReservationState.OFFERED, user=user
-    )
+    apartment_reservation.set_state(ApartmentReservationState.OFFERED, user=user)
+    update_other_customer_reservations_states(apartment_reservation)
```

Comment:
Now this makes me think we are talking about other customers 😄 But this is getting 🚲🏡  😄